graph-builder/src/config/unified.rs

```diff
@@ -0,0 +1,156 @@
+//! Unified configuration.
+
+use super::cli::CliOptions;
+use super::file::FileOptions;
+use failure::Fallible;
+use std::collections::{HashMap, HashSet};
+
+static DEFAULT_VERBOSITY: u8 = 0;
+
+static DEFAULT_SERV_ADDR: &str = "127.0.0.1";
+static DEFAULT_SERV_PORT: u16 = 8080;
+static DEFAULT_SERV_PREFIX: &str = "";
+
+static DEFAULT_REGISTRY_PERIOD: u64 = 30;
+static DEFAULT_REGISTRY_CREDENTIALS_PATH: &str = "";
+static DEFAULT_REGISTRY_MANIFESTREF_KEY: &str = "com.openshift.upgrades.graph.release.manifestref";
+static DEFAULT_REGISTRY_URL: &str = "http://localhost:5000";
+static DEFAULT_REGISTRY_REPO: &str = "openshift";
+
+static DEFAULT_STATUS_ADDR: &str = "127.0.0.1";
+static DEFAULT_STATUS_PORT: u16 = 9080;
+
+macro_rules! maybe_assign {
```

Comment:
how about `assign_if_some` here? the second `_` is emitable if you think it's too long. I think it would be a more intuitive name, as, at least for me, _maybe_ didn't tell me what the condition is.